src/main/java/com/google/googleidentity/resource/UserServlet.java

```diff
@@ -72,7 +72,7 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
         try {
             displayMainPage(request, response);
         } catch (TemplateException e) {
-            log.info("MainPage Error!");
+            log.info("MainPage Error due to TemplateException!");
```

Comment:
So here and other places where you log in a catch block, I meant actually pass `e`.

https://docs.oracle.com/javase/7/docs/api/java/util/logging/Logger.html#log(java.util.logging.Level,%20java.lang.String,%20java.lang.Throwable)

```suggestion
            log.log(Level.INFO, "MainPage Error!", e);
```